client/validators/profile.ts

```diff
@@ -40,8 +37,11 @@ export const formProfile = ({field, value, profile, errors, messages, diff}) =>
             errors[field] = gettext('Too long');
             messages.push(gettext('{{ name }} is too long', {name: fieldLabel}));
         }
-    } else if (schema.required && !schema.multilingual && (
-        typeof fieldValue === 'number' ? !fieldValue : isEmpty(fieldValue))) {
+    } else if (schema.required
+        && !schema.multilingual
+        // lodash's `isEmpty` does not handle date values correctly
```

Comment:
`isEmpty` doesn't support dates as arguments thus we shouldn't comment here about it not working as if it were a lodash issue. The way to check if date is empty is `date == null`